core/src/main/java/org/apache/calcite/sql2rel/SqlToRelConverter.java

```diff
@@ -4817,13 +4819,17 @@ void setRoot(List<RelNode> inputs) {
         if (node == null) {
```

Comment:
is node ever null? if not, remove this `if`.

Reply:
also, does `lookupExp` ever return null due to a user error? I don't think so. There's only one caller, and it throws if it receives null. So let's tighten this down, and make the function throw rather than ever returning null.